api/src/test/java/org/openmrs/module/registrationcore/api/RegistrationCoreServiceTest.java

```diff
@@ -285,4 +313,35 @@ private int getNumPatients() {
         List<List<Object>> results = adminService.executeSQL("select count(*) from patient", true);
         return ((Number)results.get(0).get(0)).intValue();
     }
+
+    @Test
+	public void importPatient_shouldSavePatientFromMpiPatient() throws Exception{
+	    MpiPatient mpiPatient = new MpiPatient();
+	    mpiPatient.addName(new PersonName("Johny", "Apple", "Smith"));
+	    Date date = new GregorianCalendar(2017, Calendar.JULY, 17).getTime();
+	    mpiPatient.setBirthdate(date);
+	    mpiPatient.setGender("M");
+	    PersonAddress personAddress = new PersonAddress();
+	    personAddress.setCountry("TesT");
+	    personAddress.setCityVillage("TeSt2");
+	    personAddress.setCountyDistrict("Test3");
+	    mpiPatient.addAddress(personAddress);
+	    PatientIdentifier identifier = new PatientIdentifier();
+	    identifier.setIdentifier("ABCD1234");
+	    identifier.setIdentifierType(patientService.getPatientIdentifierType(3));
+	    Location location = new Location();
+	    location.setId(1);
+	    identifier.setLocation(location);
+	    mpiPatient.addIdentifier(identifier);
+
+	    when(registrationCoreProperties.getMpiProvider()).thenReturn(mpiProvider);
+	    when(registrationCoreProperties.isMpiEnabled()).thenReturn(true);
+	    when(mpiProvider.fetchMpiPatient(Mockito.any(String.class), Mockito.any(String.class))).thenReturn(mpiPatient);
+	    //when(mpiProperties.getMpiPersonIdentifierTypeUuid()).thenReturn(Mockito.any(String.class))
+	    //MpiPatient mpiPatient1= serviceForMocks.findMpiPatient("abc", "efg");
+
+	    Patient patient = serviceForMocks.importMpiPatient("ABCD1234", "asdf");
+
+	    assertEquals(patient.getUuid(), mpiPatient.getUuid());
+    }
```

Comment:
The test says should save therefore I would expect to see checks that ensure that the patient didn't exist in the DB and they do after the method call.